frontend/src/metabase/entities/actions/actions.ts

```diff
@@ -1,134 +1,41 @@
 import { t } from "ttag";
-import { createAction } from "redux-actions";
 import { updateIn } from "icepick";
+import { createAction } from "redux-actions";
+
 import { createEntity } from "metabase/lib/entities";
+import { ActionsApi } from "metabase/services";
 
 import type {
-  ActionFormSettings,
-  ImplicitQueryAction,
-  WritebackActionBase,
   WritebackAction,
   WritebackActionId,
+  WritebackQueryAction,
+  WritebackImplicitQueryAction,
 } from "metabase-types/api";
 import type { Dispatch } from "metabase-types/store";
 
-import { ActionsApi } from "metabase/services";
-
-import {
-  removeOrphanSettings,
-  addMissingSettings,
-  setParameterTypesFromFieldSettings,
-  setTemplateTagTypesFromFieldSettings,
-} from "metabase/entities/actions/utils";
-import type Question from "metabase-lib/Question";
+type BaseCreateActionParams = Pick<
+  WritebackAction,
+  "name" | "description" | "model_id" | "parameters" | "visualization_settings"
+>;
 
-export type ActionParams = {
-  id?: WritebackAction["id"];
-  name: WritebackAction["name"];
-  type?: WritebackAction["type"];
-  kind?: ImplicitQueryAction["kind"];
-  description?: WritebackAction["description"];
-  model_id: WritebackAction["model_id"];
-  question?: Question;
-  formSettings?: ActionFormSettings;
+type BaseUpdateActionParams = {
+  id: WritebackAction["id"];
 };
 
-interface BaseCreateActionParams {
-  model_id: WritebackActionBase["model_id"];
-  name: WritebackActionBase["name"];
-  description: WritebackActionBase["description"];
-  parameters?: WritebackActionBase["parameters"];
-}
-
-interface UpdateActionParams {
-  id: WritebackActionBase["id"];
-}
-
-export interface CreateQueryActionOptions extends BaseCreateActionParams {
-  question: Question;
-  formSettings: ActionFormSettings;
-}
-
-export type UpdateQueryActionOptions = CreateQueryActionOptions &
-  UpdateActionParams;
-
-export interface CreateImplicitActionOptions extends BaseCreateActionParams {
-  kind: ImplicitQueryAction["kind"];
-}
+export type CreateQueryActionParams = BaseCreateActionParams &
+  Pick<WritebackQueryAction, "type" | "dataset_query">;
 
-export type UpdateImplicitActionOptions = CreateImplicitActionOptions &
-  UpdateActionParams;
+export type UpdateQueryActionParams = Partial<CreateQueryActionParams> &
+  BaseUpdateActionParams;
 
-function cleanUpQueryAction(
-  question: Question,
-  formSettings: ActionFormSettings,
-) {
-  question = setTemplateTagTypesFromFieldSettings(formSettings, question);
+export type CreateImplicitActionParams = BaseCreateActionParams &
+  Pick<WritebackImplicitQueryAction, "type" | "kind">;
```

Comment:
This is essentially `ImplicitQueryAction`?
https://github.com/metabase/metabase/blob/b591d9d36c34aa2b6933b5710dab8f0a39420745/frontend/src/metabase-types/api/actions.ts#L44-L47